src/shared/type-coercions.ts

```diff
@@ -57,35 +46,22 @@ export function toObject(value) {
 /**
  * Returns the value as a string or an empty string if the
  * value undefined, null or otherwise falsely.
- *
- * @param {any} value - initial value
- * @return {string}
  */
-export function toString(value) {
+export function toString(value: any): string {
   if (value && typeof value.toString === 'function') {
     return value.toString();
   }
   return '';
 }
 
-/**
- * @template T
- * @typedef {import('preact').Ref<T>} Ref
- */
-
 /**
  * Helper for downcasting a ref to a more specific type, where that is safe
  * to do.
  *
  * This is mainly useful to cast a generic `Ref<HTMLElement>` to a more specific
  * element type (eg. `Ref<HTMLDivElement>`) for use with the `ref` prop of a JSX element.
  * Since Preact only writes to the `ref` prop, such a cast is safe.
- *
- * @template T
- * @template {T} U
- * @param {Ref<T>|undefined} ref
- * @return {Ref<U>|undefined}
  */
-export function downcastRef(ref) {
-  return /** @type {Ref<U>|undefined} */ (ref);
+export function downcastRef<T, U>(ref: Ref<T> | undefined): Ref<U> | undefined {
+  return ref as Ref<U> | undefined;
```

Comment:
Codecov is complaining about missing coverage here. Feel free to tell Istanbul to ignore this line, since it cannot fail at runtime.